temporal-sdk/src/main/java/io/temporal/internal/nexus/NexusTaskHandlerImpl.java

```diff
@@ -177,12 +179,34 @@ private CancelOperationResponse handleCancelledOperation(
 
     OperationCancelDetails operationCancelDetails =
         OperationCancelDetails.newBuilder().setOperationId(task.getOperationId()).build();
-
-    serviceHandler.cancelOperation(ctx.build(), operationCancelDetails);
+    try {
+      serviceHandler.cancelOperation(ctx.build(), operationCancelDetails);
+    } catch (Throwable failure) {
+      convertKnownFailures(failure);
+    }
 
     return CancelOperationResponse.newBuilder().build();
   }
 
+  private void convertKnownFailures(Throwable failure) {
+    if (failure instanceof ApplicationFailure) {
+      if (((ApplicationFailure) failure).isNonRetryable()) {
+        throw new OperationHandlerException(
+            OperationHandlerException.ErrorType.BAD_REQUEST, failure.getMessage());
+      }
+      throw new OperationHandlerException(
+          OperationHandlerException.ErrorType.INTERNAL, failure.getMessage());
+    }
+    if (failure instanceof WorkflowException) {
```

Comment:
Can you confirm that this roughly maps to `WorkflowExecutionError` or `QueryRejectedError` in Go?

This is fine as an initial first pass at this but I think eventually we may want a different approach. We'll see if we get any feedback on the current approach.

Reply:
`WorkflowQueryException` and `WorkflowFailedException` extend `WorkflowException`

Reply:
So they are both treated as `BAD_REQUEST`